Homework-3-PostgreSQL-Layered-Architecture/src/dao/userDao.ts

```diff
@@ -0,0 +1,77 @@
+import { PrismaClient, User } from '@prisma/client';
+import { Service } from "typedi";
+import { BaseUserDTO, UserDTO } from "../types/userDTO";
+import { UserFilter } from "../types/filters";
+import { prismaSoftDeteleMiddleware } from '../middlewares/prismaSoftDelete';
+
+interface IUserDao {
+    findById: (id: number) => Promise<User|null>;
+    findAll: (filters: UserFilter) => Promise<User[]>;
+    create: (user:BaseUserDTO) => Promise<User>;
+    update: (user: UserDTO) => Promise<User|null>;
+    remove: (id: number) => Promise<number|null>;
+}
+
+const prisma = new PrismaClient();
```

Comment:
Not sure if it should be here. What if you have more than one entity? Will you create a client for each of them?

Reply:
Good point! Will be fixed!